clients/js/src/compat.ts

```diff
@@ -72,6 +86,17 @@ export type SapphireAnnex = {
 /** If a gas limit is not provided, the runtime will produce a very confusing error message, so we set a default limit. This one is very high, but solves the problem. This should be lowered once error messages are better or gas estimation is enabled. */
 const DEFAULT_GAS = 3_000_000;
 
+/** Web3.js */
+export function wrap<P extends AsyncSendProvider<StrictWeb3ReqArgs>>(
+  gatewayUrl: string | P,
+  customCipher?: Cipher,
+): EIP1193Provider & P & SapphireAnnex;
+/** Ethers, `window.ethereum` */
+export function wrap<U extends UpstreamProvider>(
+  upstream: U,
+  customCipher?: Cipher,
+): U & SapphireAnnex;
+
 /**
  * Wraps an upstream ethers/web3/EIP-1193 provider to speak the Sapphire format.
```

Comment:
Because of the comments above function overloads, hovering over `sapphire.wrap(new HDWalletProvider` displays unhelpful function documentation: "Web3.js"

I'd keep the longer common comment for all overloads, and keep function overloads next to function definition:
```ts
/**
 ...
 * @param customCipher An optional cipher to use for encrypting messages. If not provided an encrypting cipher will be chosen. This field is useful for providing a {@link cipher.Plain} cipher or using a custom public key for an encrypting cipher.
 */
export function wrap<P extends AsyncSendProvider<StrictWeb3ReqArgs>>( // Web3.js
  gatewayUrl: string | P,
  customCipher?: Cipher,
): EIP1193Provider & P & SapphireAnnex;
export function wrap<U extends UpstreamProvider>( // Ethers, `window.ethereum`
  upstream: U,
  customCipher?: Cipher,
): U & SapphireAnnex;
export function wrap<U extends UpstreamProvider>(
  upstream: U | string,
  customCipher?: Cipher,
): (U | (EIP1193Provider & AsyncSend)) & SapphireAnnex {
  // Already wrapped, so don't wrap it again.
  if (
```